snuba/admin/clickhouse/nodes.py

```diff
@@ -53,7 +57,8 @@ def _get_query_node(storage_key: StorageKey) -> Optional[Node]:
         query_node = cluster.get_query_node()
         return {"host": query_node.host_name, "port": query_node.port}
 
-    except (AssertionError, KeyError, UndefinedClickhouseCluster):
+    except (AssertionError, KeyError, UndefinedClickhouseCluster) as e:
+        logger.warning(str(e), extra={"storage_key": storage_key.value})
```

Comment:
My understanding is that logger.warning only hooks into sentry if I pass `exc_info=e`. Is that different for structlog?

Reply:
Structlog is unfortunately a little bit divergent from python `logging` and I think it's arguably a mistake that it tries to replicate the method names/signatures.

I don't think it's been tested but we use a special third-party library for sending structlog events to Sentry, and I think it sets the base level as warning: https://github.com/kiwicom/structlog-sentry/blob/master/structlog_sentry/__init__.py#L38

That said, it might handle strings just fine.